### PR TITLE
Rework body API

### PR DESCRIPTION
**Pull request by kelunik**

### CODE REVIEW

File: src/Connection/Http1Connection.php [hunk 1]
```diff
@@ -297,9 +297,8 @@ private function readResponse(
                 $status = $response->getStatus();
 
                 if ($status === Http\HttpStatus::SWITCHING_PROTOCOLS) {
-                    $connection = Http\parseMultipleHeaderFields($response, 'connection')[0] ?? null;
-
-                    if (!isset($connection['upgrade'])) {
+                    $connection = array_map('strtolower', Http\splitHeader($response, 'connection'));
+                    if (!\in_array('upgrade', $connection, true)) {
```
Comment (kelunik) [inline src/Connection/Http1Connection.php]: @trowski We should probably make this simpler with an API provided by `amphp/http`.

Comment (trowski) [reply src/Connection/Http1Connection.php]: Hmm, yes I didn't think about header casing. Perhaps we want to always lowercase the keys in `parseMultipleHeaderFields`? Then I think the code would look like

```php
$connection = Http\parseMultipleHeaderFields($response, 'connection')[0] ?? null;
if (isset($connection['upgrade']) {
    // ...
}
```

Comment (rekryt) [reply src/Connection/Http1Connection.php]: How about
```php
$connection = Http\parseMultipleHeaderFields($response, 'connection')[0] ?? [];
if (!\array_key_exists('upgrade', $connection)) {
    // ...
}
```

Comment (kelunik) [reply src/Connection/Http1Connection.php]: @rekryt The issue here is the casing, I think.

Comment (trowski) [reply src/Connection/Http1Connection.php]: It might be both actually. Maybe keys without a value should have an empty string as the value.

Comment (rekryt) [reply src/Connection/Http1Connection.php]: Keys without a value should have an "null" as value.
```
$connection = [ 'upgrade' => null ];
isset($connection['upgrade']) === false;
```

Comment (kelunik) [reply src/Connection/Http1Connection.php]: Updated to the newly introduced `parseHeaderTokens`.

**Comment by Bilge** [comment]
>No description provided.

lol thanks

**Pull request merged**